packages/hardhat-etherscan/src/etherscan/EtherscanService.ts

```diff
@@ -54,6 +38,40 @@ Reason: ${error.message}`,
       error
     );
   }
+
+  if (!response.ok) {
```

Comment:
I don't think this passes our linter rules. Note that the CI wasn't run in this PR because of its base branch, but otherwise it would have failed.